core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1322,6 +1326,30 @@ class KafkaApis(val requestChannel: RequestChannel,
     txnCoordinator.handleInitPid(initPidRequest.transactionalId, sendResponseCallback)
   }
 
+  def handleEndTxnRequest(request: Request): Unit = {
+    throw new UnsupportedOperationException("Not implemented yet.")
+  }
+
+  def handleAbortTxnRequest(request: Request): Unit = {
+    throw new UnsupportedOperationException("Not implemented yet.")
+  }
+
+  def handleAddPartitionToTxnRequest(request: Request): Unit = {
+    throw new UnsupportedOperationException("Not implemented yet.")
+  }
+
+  def handleAddOffsetsToTxnRequest(request: Request): Unit = {
+    throw new UnsupportedOperationException("Not implemented yet.")
+  }
+
+  def handleWriteTxnMarkerRequest(request: Request): Unit = {
+    throw new UnsupportedOperationException("Not implemented yet.")
+  }
+
+  def handleTxnOffsetCommitRequest(request: Request): Unit = {
+    throw new UnsupportedOperationException("Not implemented yet.")
```

Comment:
We can return an error. But the pattern you outlined works fine with exceptions too.

```scala
def createResponse(throttleTimeMs: Int: AbstractResponse = throw new UnsupportedOperationException("Not yet implemented")
sendResponseMaybeThrottle(request, createResponse)
```